XenModel/Actions/Updates/DownloadAndUnzipXenServerPatchAction.cs

```diff
@@ -60,17 +60,19 @@ public string PatchPath
             get; private set;
         }
 
-        public DownloadAndUnzipXenServerPatchAction(string patchName, Uri uri, string outputFileName)
-            : this(patchName, uri, outputFileName, InvisibleMessages.XEN_UPDATE)
-        { }
+        private bool DownloadedUpdateFromUri
+        {
+            get { return address != null; }
+        }
 
-        public DownloadAndUnzipXenServerPatchAction(string patchName, Uri uri, string outputFileName, string updateFileExtension)
-            : base(null, string.Format(Messages.DOWNLOAD_AND_EXTRACT_ACTION_TITLE, patchName), string.Empty, false)
+        public DownloadAndUnzipXenServerPatchAction(string patchName, Uri uri, string outputFileName, bool suppressHist, params string[] updateFileExtensions)

```

Comment:
Just a note on the name of this method: DownloadAndUnzipXenServerPatchAction suggests that it returns an action, when in fact it creates and execute an action. A better name would be DownloadAndUnzipUpdate,